.github/workflows/cd-internal-testnet.yml

```diff
@@ -1,60 +1,45 @@
+
+## run after every successful CI job of new commits to the master branch, or on manual dispatch
 on:
   workflow_run:
     workflows: [Continuous Integration (Kava Master)]
     types:
       - completed
+  workflow_dispatch:
 
 jobs:
+  # when not manually dispatched, we only want merges to master that change the relevant files
+  # to trigger the network reset
   changed_files:
     runs-on: ubuntu-latest
-    # define output for first job forwarding output of changedInternalTestnetConfig job
     outputs:
-        changedInternalTestnetConfig: ${{ steps.changed-internal-testnet-config.outputs.any_changed }}
+      changedInternalTestnetConfig: ${{ steps.changed-internal-testnet-config.outputs.any_changed }}
     steps:
-        - uses: actions/checkout@v4
-          with:
-            fetch-depth: 0 # OR "2" -> To retrieve the preceding commit.
-        - name: Get all changed internal testnet files
-          id: changed-internal-testnet-config
-          uses: tj-actions/changed-files@v42
-          with:
-            # Avoid using single or double quotes for multiline patterns
-            files: |
-              ci/env/kava-internal-testnet/**
-
-  # in order:
-  # enter standby (prevents autoscaling group from killing node during deploy)
-  # stop kava
-  # take ebs + zfs snapshots
-  # download updated binary and genesis
-  # reset application database state (only done on internal testnet)
-  reset-chain-to-zero-state:
-    needs: [changed_files]
-    # only start cd pipeline if last ci run was successful
-    if: ${{ github.event.workflow_run.conclusion == 'success' && needs.changed_files.outputs.changedInternalTestnetConfig == 'true' }}
-    uses: ./.github/workflows/cd-reset-internal-testnet.yml
-    with:
-      aws-region: us-east-1
-      chain-id: kava_2221-17000
-      ssm-document-name: kava-testnet-internal-node-update
-      playbook-name: reset-internal-testnet-playbook.yml
-      playbook-infrastructure-branch: master
-    secrets: inherit
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # OR "2" -> To retrieve the preceding commit.
+      - name: Get all changed internal testnet files
+        id: changed-internal-testnet-config
+        uses: tj-actions/changed-files@v42
+        with:
+          # Avoid using single or double quotes for multiline patterns
+          files: |
+            ci/env/kava-internal-testnet/**
 
-  # start kava with new binary and genesis state on api, peer and seed nodes, place nodes in service once they start and are synched to live
-  start-chain-api:
-    uses: ./.github/workflows/cd-start-chain.yml
+  reset-and-restart-chain:
+    # only start cd pipeline if last ci run was successful (or it this is being manually dispatched)
+    if: |
+      github.event_name == 'workflow_dispatch' ||
+      (github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'success' && needs.changed_files.outputs.changedInternalTestnetConfig == 'true')
```

Comment:
this is the logic for whether or not to actually reset the network.

it amounts to:
1. reset network if we are manually dispatching the flow. the manual dispatch is a specific choice. if i hit the reset network button, the network should reset regardless of what the content of the HEAD commit is.
2. when a commit to master passes the test CI, reset the network if the files in `ci/kava-internal-testnet` have changed. _only commits that include changes to `KAVA.VERSION` or the genesis file trigger an automated reset_